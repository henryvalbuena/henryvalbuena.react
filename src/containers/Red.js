import React from "react";

import aboutMe from "../about";
import projects from "../projectList";
import menuItems from "../constants";

import styles from "./Red.module.css";

function Red(props) {
  return (
    <div className={styles["container"]}>
      <div className={styles["header-red"]}>
        <h1 className={styles["desktop-h1"]}>Welcome To The Rabbit Hole</h1>
        <h1 className={styles["mobile-h1"]}>Welcome</h1>
        <hr />
      </div>
      <div className={styles["main"]}>
        <div className={styles["text-container"]}>
          <p>{aboutMe.intro}</p>
        </div>
        <div className={styles["text-container"]}>
          <h2 className={styles["h2-headers"]}>Interests</h2>
          <ul>
            {aboutMe.interest.map((ele, i) => (
              <li key={i}>{ele}</li>
            ))}
          </ul>
        </div>
        <div className={styles["text-container"]}>
          <h2 className={styles["h2-headers"]}>Projects</h2>
          <div className={styles["projects"]}>
            {projects.map((p, i) => (
              <div key={i} className={styles["project"]}>
                <a
                  className={styles["project-a"]}
                  href={p.src}
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  {p.name}
                </a>
              </div>
            ))}
          </div>
        </div>
      </div>
      <div className={styles["footer-desktop"]}>
        {menuItems.social.map((ele, i) => (
          <a key={i} href={ele.href} target="_blank" rel="noopener noreferrer">
            <i className={ele.icon}></i>
          </a>
        ))}
      </div>
      <div className={styles["footer-mobile"]}>
        {menuItems.social.map((ele, i) => (
          <a key={i} href={ele.href} target="_blank" rel="noopener noreferrer">
            <i className={ele.icon}></i>
          </a>
        ))}
      </div>
    </div>
  );
}

export default Red;
