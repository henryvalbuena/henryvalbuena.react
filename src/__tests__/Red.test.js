import React from "react";
import { render } from "@testing-library/react";
import Red from "../containers/Red";

describe("Renders", () => {
  it("Has <h1> tags", () => {
    const { getByText } = render(<Red />);
    expect(getByText("Welcome")).toBeInTheDocument();
    expect(getByText("Welcome To The Rabbit Hole")).toBeInTheDocument();
  });

  it("Has Projects", () => {
    const { getByText } = render(<Red />);
    expect(getByText("Projects")).toBeInTheDocument();
  });

  it("Has Interests", () => {
    const { getByText } = render(<Red />);
    expect(getByText("Interests")).toBeInTheDocument();
  });
});

it("Snapshot Red", () => {
  expect(render(<Red />)).toMatchSnapshot();
});
