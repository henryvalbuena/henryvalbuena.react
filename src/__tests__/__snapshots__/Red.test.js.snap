// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Red 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="header-red"
        >
          <h1
            class="desktop-h1"
          >
            Welcome To The Rabbit Hole
          </h1>
          <h1
            class="mobile-h1"
          >
            Welcome
          </h1>
          <hr />
        </div>
        <div
          class="main"
        >
          <div
            class="text-container"
          >
            <p>
              Hi, I am Henry and I program computers and systems to make life of people easier or just simply to have fun. If you want to know more hit me with an email or checkout my github.
            </p>
          </div>
          <div
            class="text-container"
          >
            <h2
              class="h2-headers"
            >
              Interests
            </h2>
            <ul>
              <li>
                Blockchain
              </li>
              <li>
                Performance and Optimization
              </li>
              <li>
                Full-stack Development
              </li>
              <li>
                Functional training and programming ;)
              </li>
              <li>
                Nature
              </li>
              <li>
                Learning
              </li>
              <li>
                The Matrix
              </li>
            </ul>
          </div>
          <div
            class="text-container"
          >
            <h2
              class="h2-headers"
            >
              Projects
            </h2>
            <div
              class="projects"
            >
              <div
                class="project"
              >
                <a
                  class="project-a"
                  href="https://github.com/henryvalbuena/full-stack-project"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Udacity Latte Cafe
                </a>
              </div>
              <div
                class="project"
              >
                <a
                  class="project-a"
                  href="https://github.com/henryvalbuena/latte-machine"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Latte-Machine
                </a>
              </div>
              <div
                class="project"
              >
                <a
                  class="project-a"
                  href="https://github.com/henryvalbuena/SpaceBattle"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Desktop Web Browser Game
                </a>
              </div>
              <div
                class="project"
              >
                <a
                  class="project-a"
                  href="https://github.com/henryvalbuena/chatbot"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Chat
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="footer-desktop"
        >
          <div>
            <a
              href="https://www.linkedin.com/in/henry-valbuena/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="briefcase icon"
              />
            </a>
          </div>
          <div>
            <a
              href="https://github.com/henryvalbuena"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="github icon"
              />
            </a>
          </div>
          <div>
            <a
              href="mailto:henry.valbuena@outlook.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="mail icon"
              />
            </a>
          </div>
        </div>
        <div
          class="footer-mobile"
        >
          <div>
            <a
              href="https://www.linkedin.com/in/henry-valbuena/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="briefcase icon"
              />
            </a>
          </div>
          <div>
            <a
              href="https://github.com/henryvalbuena"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="github icon"
              />
            </a>
          </div>
          <div>
            <a
              href="mailto:henry.valbuena@outlook.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                class="mail icon"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="header-red"
      >
        <h1
          class="desktop-h1"
        >
          Welcome To The Rabbit Hole
        </h1>
        <h1
          class="mobile-h1"
        >
          Welcome
        </h1>
        <hr />
      </div>
      <div
        class="main"
      >
        <div
          class="text-container"
        >
          <p>
            Hi, I am Henry and I program computers and systems to make life of people easier or just simply to have fun. If you want to know more hit me with an email or checkout my github.
          </p>
        </div>
        <div
          class="text-container"
        >
          <h2
            class="h2-headers"
          >
            Interests
          </h2>
          <ul>
            <li>
              Blockchain
            </li>
            <li>
              Performance and Optimization
            </li>
            <li>
              Full-stack Development
            </li>
            <li>
              Functional training and programming ;)
            </li>
            <li>
              Nature
            </li>
            <li>
              Learning
            </li>
            <li>
              The Matrix
            </li>
          </ul>
        </div>
        <div
          class="text-container"
        >
          <h2
            class="h2-headers"
          >
            Projects
          </h2>
          <div
            class="projects"
          >
            <div
              class="project"
            >
              <a
                class="project-a"
                href="https://github.com/henryvalbuena/full-stack-project"
                rel="noopener noreferrer"
                target="_blank"
              >
                Udacity Latte Cafe
              </a>
            </div>
            <div
              class="project"
            >
              <a
                class="project-a"
                href="https://github.com/henryvalbuena/latte-machine"
                rel="noopener noreferrer"
                target="_blank"
              >
                Latte-Machine
              </a>
            </div>
            <div
              class="project"
            >
              <a
                class="project-a"
                href="https://github.com/henryvalbuena/SpaceBattle"
                rel="noopener noreferrer"
                target="_blank"
              >
                Desktop Web Browser Game
              </a>
            </div>
            <div
              class="project"
            >
              <a
                class="project-a"
                href="https://github.com/henryvalbuena/chatbot"
                rel="noopener noreferrer"
                target="_blank"
              >
                Chat
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="footer-desktop"
      >
        <div>
          <a
            href="https://www.linkedin.com/in/henry-valbuena/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="briefcase icon"
            />
          </a>
        </div>
        <div>
          <a
            href="https://github.com/henryvalbuena"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="github icon"
            />
          </a>
        </div>
        <div>
          <a
            href="mailto:henry.valbuena@outlook.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="mail icon"
            />
          </a>
        </div>
      </div>
      <div
        class="footer-mobile"
      >
        <div>
          <a
            href="https://www.linkedin.com/in/henry-valbuena/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="briefcase icon"
            />
          </a>
        </div>
        <div>
          <a
            href="https://github.com/henryvalbuena"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="github icon"
            />
          </a>
        </div>
        <div>
          <a
            href="mailto:henry.valbuena@outlook.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <i
              class="mail icon"
            />
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
